challenge_4/python/OkraArmband/src/challenge_4.py

```diff
@@ -0,0 +1,49 @@
+class BinaryTree():
+    def __init__(self, root):
+        self.root = root
+        self.left = None
+        self.right = None
+
+    def invertTree(self):
+        """
+        Swap left and right children of input BinaryTree for all subsequent children as well (recursively).
+        """
+        temp = self.left
+        self.left = self.right
+        self.right = temp
+
+
+        # Continue the recursion as long as we haven't reached the end of a branch
+        if self.left is not None:
+            self.left.invertTree()
+        if self.right is not None:
+            self.right.invertTree()
+
+    @classmethod
+    def from_list(cls,data):
```

Comment:
Yeah, I wanted to make this generalizable so that you could use any type of data type in the sort of structure where you have a "main" thing connected to two children. BSTs limit this to numbers, so I guess the thing to do if you're trying to enforce that is just throw an error message in the constructor if they pass in a list, right?